tests/test_pretest.py

```diff
@@ -294,3 +294,8 @@ def test_generate_running_golden_config(duthosts):
     """
     for duthost in duthosts:
         duthost.shell("sonic-cfggen -d --print-data > /etc/sonic/running_golden_config.json")
+        if duthost.is_multi_asic:
+            for asic_index in range(0, duthost.facts.get('num_asic')):
+                asic_ns = 'asic{}'.format(asic_index)
```

Comment:
Could use duthost.get_namespace_from_asic_id(asic_index)

Reply:
We can't use get_namespace_from_asic_id, as this is relying on asics that are present in the card. For a chassis with not all fabric cards, the supervisor card will only have asics for the fabric cards that are present in the 'asics' variable. For asic_index of asics that are not present, we get an error